build.py

```diff
@@ -173,6 +173,10 @@ def __init__(self, name, env):
             if env.TargetOSIs('darwin'):
                 self.add_icecream_check()
             self.set_up_icecc()
+        if env.get('_NINJA_NO_TEST_EXECUTION'):
```

Comment:
nit: `self.setup_test_execution = '_NINJA_NO_TEST_EXECUTION' not in env` or `not env.get('_NINJA_NO_TEST_EXECUTION')` if you expect it to be present but set to something Falsey.

This feels a lot like my pet peeve of
```py
if cond:
  return True
else:
  return False
```

Reply:
Done.